### PR TITLE
[SearchSnippet][Part-1] Implement SearchSnippet/get

### PR DESCRIPTION
**Pull request by tddang-linagora**
## Related issue
- https://github.com/linagora/tmail-flutter/issues/3234

### CODE REVIEW

**Review by dab246 — APPROVED** [review]
- [ ] Please rebase

File: lib/jmap/mail/email/search_snippet/search_snippet_get_response.dart [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import 'package:json_annotation/json_annotation.dart';
+
+part 'search_snippet_get_response.g.dart';
+
+@JsonSerializable(
+  includeIfNull: false,
+  explicitToJson: true,
+  converters: [
+    AccountIdConverter(),
+    EmailIdConverter(),
```
Comment (hoangdat) [inline lib/jmap/mail/email/search_snippet/search_snippet_get_response.dart:16]: why we need this converter?

File: test/jmap/mail/email/search_snippet/search_snippet_get_method_test.dart [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+            "path": "/ids/*"
+          },
+        },
+        "c1"
+      ]
+    ]
+  };
+
+  group('search snippet get method test:', () {
+    test(
```
Comment (hoangdat) [inline test/jmap/mail/email/search_snippet/search_snippet_get_method_test.dart:92]: can you test the case if searchSnippet error

File: lib/jmap/mail/email/search_snippet/search_snippet_get_response.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/jmap/mail/email/search_snippet/search_snippet_get_response.dart]: Removed

File: test/jmap/mail/email/search_snippet/search_snippet_get_method_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/jmap/mail/email/search_snippet/search_snippet_get_method_test.dart:94]: Done

**Review by hoangdat — APPROVED** [review]

**Review by dab246 — APPROVED** [review]
- [ ] Please rebase it

**Pull request merged**